lexxpluss_apps/src/towing_unit_controller.cpp

```diff
@@ -0,0 +1,184 @@
+#include <device.h>
+#include <drivers/gpio.h>
+#include <logging/log.h>
+#include <shell/shell.h>
+#include "towing_unit_controller.hpp"
+
+namespace lexxhard::towing_unit_controller {
+
+LOG_MODULE_REGISTER(towing_unit);
+
+char __aligned(4) msgq_towing_unit_status_buffer[8 * sizeof (msg_towing_unit_status)];
+char __aligned(4) msgq_towing_unit_power_on_buffer[8 * sizeof (msg_towing_unit_status)];
+
+class towing_unit_controller_impl {
+public:
+    int init() {
+        k_msgq_init(&msgq_towing_unit_status, msgq_towing_unit_status_buffer, sizeof (msg_towing_unit_status), 8);
+        k_msgq_init(&msgq_towing_unit_power_on, msgq_towing_unit_power_on_buffer, sizeof (msg_towing_unit_status), 8);
+
+        const device *gpioj{device_get_binding("GPIOJ")};
+        if (device_is_ready(gpioj)) {
+            gpio_pin_configure(gpioj, 1, GPIO_OUTPUT);  //Power ON Output SPRGPIO4
+            gpio_pin_configure(gpioj, 2, GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_HIGH);   //Switch 1 SPRGPIO5
+            gpio_pin_configure(gpioj, 3, GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_HIGH);   //Switch 2 SPRGPIO6
+            gpio_pin_configure(gpioj, 4, GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_HIGH);   //PowerGood +12V SPRGPIO7
+        }
+
+        gpio_pin_set(gpioj, 1, 0);  //Set 12V Power ON(active low)
+        is_towing_unit_power_on = V12_ON;
+
+        // Debug LED5 is ON
+        const device *gpiog{device_get_binding("GPIOG")};
+        if (device_is_ready(gpiog)) {
+            gpio_pin_configure(gpiog, 3, GPIO_OUTPUT_HIGH);
+        }
+
+        LOG_INF("Towing Unit Init Done");
+
+        return 0;
+    }
+    void run() {
+        LOG_INF("Towing Unit run() started");
+        
+        while (true) {
+            //Get Switch & Power Good Status
+            const device *gpioj{device_get_binding("GPIOJ")};
+
+            if (device_is_ready(gpioj)) {
+                if (is_towing_unit_power_on == V12_ON) {
+                    gpio_pin_set(gpioj, 1, 0);  //Set 12V Power ON(active low)
+                } else {
+                    gpio_pin_set(gpioj, 1, 1);  //Set 12V Power OFF
+                }
+            }
+            if (device_is_ready(gpioj)) {
+                /* SW_L */ 
+                if(gpio_pin_get(gpioj, 2) == 0){
+                    is_towing_unit_sw_l_loading = LOADED;  //Loading
+                }else{
+                    is_towing_unit_sw_l_loading = UNLOADED; //Not Loading
+                }
+            }
+            if (device_is_ready(gpioj)) {
+                /* SW_R */ 
+                if(gpio_pin_get(gpioj, 3) == 0){
+                    is_towing_unit_sw_r_loading = LOADED;  //Loading
+                }else{
+                    is_towing_unit_sw_r_loading = UNLOADED; //Not Loading
+                }
+            }
+            if (device_is_ready(gpioj)) {
+                /* Power Good */ 
+                if(gpio_pin_get(gpioj, 4) == 0){
+                    is_towing_unit_power_good = V12_OK;  //+12V is on
```

Comment:
```suggestion
                    is_towing_unit_power_good = V12_ON;
```

Reply:
最新コミットにて修正済み。

Reply:
I had commented from the comment that I thought it would be appropriate to use V12_ON instead of V12_OK for the definition to be used. Currently it is still V12_OK, but after reading the code again, I thought V12_OK seemed to be the right choice.

使用する定義をコメントから、V12_OKではなくV12_ONとするのが適切かと思ってコメントしていました。現状V12_OKのままですが、コードを読み返すとV12_OKで良さそうに思いました。